lib/SendStream.js

```diff
@@ -226,6 +224,74 @@ SendStream.prototype.isConditionalGET = function isConditionalGET () {
     this.req.headers['if-modified-since']
 }
 
+SendStream.prototype.isNotModifiedFailure = function isNotModifiedFailure () {
+  const req = this.req
+  const res = this.res
+
+  // Always return stale when Cache-Control: no-cache
+  // to support end-to-end reload requests
+  // https://tools.ietf.org/html/rfc2616#section-14.9.4
+  if (
+    'cache-control' in req.headers &&
+    req.headers['cache-control'].indexOf('no-cache') !== -1
+  ) {
+    return false
+  }
+
+  // if-none-match
+  if ('if-none-match' in req.headers) {
+    const ifNoneMatch = req.headers['if-none-match']
+
+    if (ifNoneMatch === '*') {
+      return true
+    } else {
```

Comment:
we return anyway, else is redundant

Reply:
If I remove this return true, the test "where "If-None-Match" is set" "should respond with 306 Not Modified when using wildcard * on existing file" will fail.

Reply:
return is fine, else is unnecessary

Reply:
ah right